tests/v2/validation/fleet/provisioning/airgap/fleet_in_airgap_test.go

```diff
@@ -0,0 +1,224 @@
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(a.T(), err)
+
+	a.client = client
+
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	enabled := true
+
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(a.T(), err)
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(a.T(), err)
+
+	a.standardClient = standardUserClient
+
+	corralConfig := corral.Configurations()
+	err = corral.SetupCorralConfig(corralConfig.CorralConfigVars, corralConfig.CorralConfigUser, corralConfig.CorralSSHPath)
+	require.NoError(a.T(), err)
+
+	a.corralPackage = corral.PackagesConfig()
+
+	bastionIP := corralConfig.CorralConfigVars[corralBastionIP]
+	bastionUser := corralConfig.CorralConfigVars[corralSSHUser]
+
+	// format the privateKey string for use with ssh package.
+	privateKey := corralConfig.CorralConfigVars[corralPrivateKey]
+	privateKey = strings.Replace(privateKey, "\\n", "\n", -1)
+	privateKey = strings.Replace(privateKey, "\"", "", -1)
+	privateKey = fmt.Sprintf("%s", privateKey)
+
+	err = corral.UpdateCorralConfig(corralBastionIP, bastionIP)
+	require.NoError(a.T(), err)
+
+	registrySetting, err := client.Management.Setting.ByID(systemRegistry)
+	require.NoError(a.T(), err)
+
+	a.registryFQDN = registrySetting.Value
+
+	internalIP, err := setupAirgapFleetResources(bastionUser, bastionIP, privateKey)
+	require.NoError(a.T(), err, "failed to setup local git repo on bastion")
+
+	fleetSecretName, err := createFleetSSHSecret(client, privateKey)
+	require.NoError(a.T(), err, "failed to create SSH Secrets for fleet")
+
+	a.fleetGitRepo = &v1alpha1.GitRepo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fleet.FleetMetaName + namegenerator.RandStringLower(5),
+			Namespace: fleet.Namespace,
+		},
+		Spec: v1alpha1.GitRepoSpec{
+			Repo:            fmt.Sprintf("%s@%s:/home/%s/%s", bastionUser, internalIP, bastionUser, fleetExampleFolderName),
+			Branch:          fleet.BranchName,
+			Paths:           []string{fleet.GitRepoPathLinux},
+			CorrectDrift:    &v1alpha1.CorrectDrift{},
+			ImageScanCommit: v1alpha1.CommitSpec{AuthorName: "", AuthorEmail: ""},
+			Targets: []v1alpha1.GitTarget{
+				{
+					ClusterSelector: &metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{
+								Key:      fleet.MatchKey,
+								Operator: fleet.MatchOperator,
+								Values: []string{
+									fleet.HarvesterName,
+								},
+							},
+						},
+					},
+				},
+			},
+			ClientSecretName: fleetSecretName,
+		},
+	}
+
+	a.client, err = a.client.ReLogin()
+	require.NoError(a.T(), err)
+
+	if a.clustersConfig.RKE2KubernetesVersions == nil {
+		rke2Versions, err := kubernetesversions.ListRKE2AllVersions(a.client)
```

Comment:
I noticed the struct is named rke2 as well. why is this specific to rke2 clusters and why not both rKE2 and k3s?

Reply:
since this test is geared toward testing fleet, the cluster type doesn't really matter. I decided to use rke2 across all the fleet/provisioning tests for consistency.

Reply:
the test is using rke2 though. I'm not sure what you mean but the test is structured to use RKE2, as there's no real reason to support all 3 default engines when testing fleet. Does that make more sense?